### ISSUE TITLE
Brickognize Camera screen is black but still works. V 2024.2.1 Build 1184

### ISSUE THREAD

**Issue by Brick-Slayer** [rgriebl/brickstore#845]
When I click the camera icon to open the brickognize feature the preview screen is black and you cannot see the part. But if i click the spacebar it will open the correct part in brickstore so the brickognize feature is working but the display is just black for the camera. Seems this has been an issue on Macs for the last few versions. If I revert back a few versions it will work again.

Currents system:
Mac Book Pro
13-inch, M1, 2020
Chip Apple M1
Memory 8GB
Mac Os Sonoma 14.2.1

![Screenshot 2024-03-06 at 11 06 26 AM](https://github.com/rgriebl/brickstore/assets/77291704/9d40ce14-fe53-4346-af2e-fd373a8279b7)

### BrickStore 2024.2.1 (build: 1184)

* **brickstore.ldraw**: 13 Feb 2024 11:33:19 -0600
* **brickstore.locale**: en
* **build.arch**: x86_64
* **build.compiler**: clang 14.0.0 (clang-1400.0.29.202)
* **build.date**: 2024-02-14T17:37:09.000
* **build.host**: Mac-1707931639775.local
* **build.number**: 1184
* **build.qt.version**: 6.4.3
* **hw.cpu**: Apple M1
* **hw.cpu.arch**: x86_64
* **hw.gpu**: Apple M1
* **hw.gpu.arch**: sppci_vendor_Apple
* **hw.memory.gb**: 8.0
* **os.arch**: x86_64
* **os.productname**: macOS 14.2
* **qt.debug**: false
* **qt.platformname**: cocoa
* **qt.version**: 6.4.3

**Comment by sirlaugher** [comment]
I have this problem as well and am on a Windows machine. I have tried this on another PC I have that is Windows and have the same issue.

**Comment by pchtri** [comment]
Same issue here on Windows 11. The black camera problem appeared a couple of versions ago.
Preview works fine if I use Brickognize.com

**Comment by BrickerMom** [comment]
I'm experiencing the same issue. When I connected the camera for the first time, BrickStore displayed the camera view correctly. When I relaunched BrickStore the next day the camera display was showing black. Like the others, the camera is still capturing the image and Brickognize is finding the part.

Mac Mini(late 2014), running Monterey,  BrickStore version 2024.2.1 (build:1184)

**Comment by rgriebl** [comment]
I finally managed to reproduce this problem once now on a Mac, but not in a state where I could debug it.
What did work for me, was to simply resize the window.

Could you please try if this works for you too?

**Comment by pchtri** [comment]
Yes! Resizing the window did resolve the problem for me on Windows 11.

**Comment by BrickerMom** [comment]
Here's the crazy thing. When I first opened the camera yesterday I was
getting the black screen view. Then about half way through the day it
started showing the image. I didn't do anything that I can figure out. I
didn't update anything, I didn't move the window, I didn't change the size,
nothing.  It was showing black 1 minute and an image the next.

Ginny

**Comment by Brick-Slayer** [comment]
I downloaded the new version again and now it seems to be working fine no matter what size the window it. Thanks for your help in solving this issue on Mac.

**Comment by rgriebl** [comment]
For people on Macs: the problem comes from the "macOS-10-Legacy" (**) version of BrickStore. It has known issues with Webcam access, but I cannot fix them anymore, as macOS 10.x is not supported anymore.
Now the auto update had a bug in the past, where it sometimes installed the legacy version on up-to-date Macs. This will be rectified by the next auto update, or you can manually download the normal version and replace the application (this is what @Brick-Slayer did).

As for the Windows 11 problem: I have no idea right now, if you are on the latest BrickStore version 2024.2.1. If you are on an older version, that is a known - and fixed - bug.

(**) Go to `Help` > `System Information`: if it says `build.qt.version: 6.4.3`, you are using the legacy macOS 10 version.

**Comment by pilotflying777** [comment]
> I have this problem as well and am on a Windows machine. I have tried this on another PC I have that is Windows and have the same issue.

Same for me. I am using the latest version 2024.3.1 on Windows 11 and the screen remains black until I resize the window manually. Please have a look.

**Comment by henriquedesousa** [comment]
Hello, I have the same issue, but I'm on Linux Mint (Ubuntu 22.04 based). I have libqt6multimedia6 installed. 

> sudo apt-get install libqt6multimedia6
> Reading package lists... Done
> Building dependency tree... Done
> Reading state information... Done
> libqt6multimedia6 is already the newest version (6.2.4-1).
 
I have tried to resize the window, but without success. There's some useful log entries when I run Bickstore from the command line:


```
qrc:/Scanner/CameraPreview.qml:6:1: module "QtMultimedia" is not installed 
     import QtMultimedia 
     ^ (at qrc:/Scanner/CameraPreview.qml, 6)
CameraPreviewWidget: failed to load QML file: (at /brickstore/src/scanner/camerapreviewwidget.cpp, 33)
   "qrc:/Scanner/CameraPreview.qml:6:1: module \"QtMultimedia\" is not installed" (at /brickstore/src/scanner/camerapreviewwidget.cpp, 36)
NEW STATE Scanner::Capture::State::Inactive [[ OLD STATE Scanner::Capture::State::Idle ]] (at /brickstore/src/scanner/capture.cpp, 296)
```


But the incredible thing is.. is that the recognition is actually working ! It's just very difficult to center a part on the webcam without the preview. 

Vielen Dank für Ihr Hilfe.
